docs: Update docs to run the pipeline (#4379)
DFE-Digital/education-benchmarking-and-insights: data-pipeline/README.md

    * `poetry install`

### Running the pipeline locally

> > **Warning:** These instructions run the pipeline **locally only**, against the Azurite emulator. Do not use them to run the pipeline on a deployed environment. To run the pipeline, follow the [Running the pipeline doc](../documentation/data/05_Releases.md), which triggers the run through the `data-pipeline-job-pending` queue and the Orchestrator. Triggering a deployed run any other way (for example, posting directly to `data-pipeline-job-default-start`) bypasses the Orchestrator and risks the service not operating correctly.

Reviewer: ```suggestion
> **Warning:** These instructions run the pipeline **locally only**, against the Azurite emulator. Do not use them to run the pipeline on a deployed environment. To run the pipeline, follow the [Running the pipeline doc](../documentation/data/05_Releases.md), which triggers the run through the `data-pipeline-job-pending` queue and the Orchestrator. Triggering a deployed run by posting directly to `data-pipeline-job-default-start` bypasses the Orchestrator and the run doesn't get tagged with a `JobId` in the database.
```
The JobId is used by the frontend to retrieve user-specific custom data runs from the database.

Reviewer 2: @murdo-hippo the job id isn't used for that purpose. The JobId is used to correlated orchestrations. The front-end does not create the JobId.

The message placed in the pending queue will never have a JobId (default & custom job). It is added by the orchestrator.

Reviewer 2: @Faizan-ahmad00 you need to revert the change as it's misleading.

Author: reverted the changes as suggested and kept it generic rather than specifying the exact issue.
